### PR TITLE
feat: label system socket and runtime files

### PR DESCRIPTION
**Pull request by dsseng**
Set SELinux labels so that services could gain access permissions.

### CODE REVIEW

**Review by smira — APPROVED** [review]

File: internal/app/apid/main.go [hunk 1]
```diff
@@ -157,6 +158,10 @@ func apidMain() error {
 		return fmt.Errorf("error creating listner: %w", err)
 	}
 
+	if err = selinux.SetLabel(constants.APISocketPath, "system_u:object_r:apid_socket_t:s0"); err != nil {
```
Comment (frezbo) [inline internal/app/apid/main.go:161]: constant maybe?

File: internal/app/machined/pkg/system/services/apid.go [hunk 2]
```diff
@@ -96,6 +97,10 @@ func (o *APID) PreFunc(ctx context.Context, r runtime.Runtime) error {
 		return err
 	}
 
+	if err := selinux.SetLabel(constants.APIRuntimeSocketPath, "system_u:object_r:apid_runtime_socket_t:s0"); err != nil {
```
Comment (dsseng) [inline internal/app/machined/pkg/system/services/apid.go:100]: Constant

File: internal/pkg/logind/broker.go [hunk 3]
```diff
@@ -48,11 +50,19 @@ func NewBroker(serviceSocketPath, clientSocketPath string) (*DBusBroker, error)
 		return nil, err
 	}
 
+	if err = selinux.SetLabel(serviceSocketPath, "system_u:object_r:dbus_service_socket_t:s0"); err != nil {
```
Comment (frezbo) [inline internal/pkg/logind/broker.go:53]: should we do similar like standard distro, `system_dbus_var_run_t` :thinking:

File: internal/pkg/selinux/policy/selinux/services/cri.cil [hunk 4]
```diff
@@ -15,3 +15,30 @@
 (call pod_p (etcd_t))
 ; FIXME: insecure as anyone with access to the pod containerd may obtain this domain (executable in ephemeral)
```
Comment (frezbo) [inline internal/pkg/selinux/policy/selinux/services/cri.cil:16]: is the comment still valid?

Comment (dsseng) [reply internal/pkg/selinux/policy/selinux/services/cri.cil:16]: Yes. Ephemeral label actually comes later in the series, and even after then we need to think how to label kubelet/etcd executables and assign labels to static pods

File: internal/pkg/logind/broker.go (hunk 3 above)
Comment (dsseng) [reply internal/pkg/logind/broker.go:53]: Unsure. No reason to either do as in reference policy nor do it differently. We can do as in ref though

File: internal/pkg/selinux/policy/selinux/services/cri.cil (hunk 4 above)
Comment (frezbo) [reply internal/pkg/selinux/policy/selinux/services/cri.cil:16]: okay, let's keep what we have then, unless we figure we need to relabel it later

File: internal/pkg/logind/broker.go (hunk 3 above)
Comment (frezbo) [reply internal/pkg/logind/broker.go:53]: let keep what you have then.

File: internal/integration/api/selinux.go [hunk 5]
```diff
@@ -64,6 +67,92 @@ func (suite *SELinuxSuite) getLabel(nodeCtx context.Context, pid int32) string {
...
+	suite.checkFileLabels(workers, expectedLabelsWorker)
+	suite.checkFileLabels(controlplanes, expectedLabelsControlPlane)
+}
+
+func (suite *SELinuxSuite) checkFileLabels(nodes []string, expectedLabels map[string]string) {
+	for _, node := range nodes {
+		nodeCtx := client.WithNode(suite.ctx, node)
+		cmdline := suite.ReadCmdline(nodeCtx)
+
+		seLinuxEnabled := pointer.SafeDeref(procfs.NewCmdline(cmdline).Get(constants.KernelParamSELinux).First()) != ""
```
Comment (frezbo) [inline internal/integration/api/selinux.go:111]: this can be moved in setuptest so we can skip earlier

**Review by frezbo — APPROVED** [review]

File: internal/integration/api/selinux.go (hunk 5 above)
Comment (dsseng) [reply internal/integration/api/selinux.go:111]: I do it here to check per-node, as we might have different cmdline on each node

**Pull request merged**